Mage.Sets/src/mage/cards/a/ArmedAndArmored.java

```diff
@@ -0,0 +1,123 @@
+
+    public ArmedAndArmoredEffect() {
+        super(Duration.EndOfTurn, Layer.TypeChangingEffects_4, SubLayer.NA, Outcome.BecomeCreature);
+        staticText = "Vehicles you control become artifact creatures until end of turn";
+    }
+
+    private ArmedAndArmoredEffect(final ArmedAndArmoredEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ArmedAndArmoredEffect copy() {
+        return new ArmedAndArmoredEffect(this);
+    }
+
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
+        for (Permanent permanent : game.getBattlefield().getAllActivePermanents(source.getControllerId())) {
+            if (permanent != null && permanent.hasSubtype(SubType.VEHICLE, game)) {
+                if (sublayer == SubLayer.NA) {
+                    permanent.addCardType(CardType.ARTIFACT);
+                    permanent.addCardType(CardType.CREATURE);// TODO: Check if giving CREATURE Type is correct
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return false;
+    }
+}
+
+class ArmedAndArmoredEquipEffect extends OneShotEffect {
+
+    public ArmedAndArmoredEquipEffect() {
+        super(Outcome.Benefit);
+        staticText = "Choose a Dwarf you control. Attach any number of Equipment you control to it.";
+    }
+
+    private ArmedAndArmoredEquipEffect(final ArmedAndArmoredEquipEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ArmedAndArmoredEquipEffect copy() {
+        return new ArmedAndArmoredEquipEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        FilterControlledCreaturePermanent dwarfFilter = new FilterControlledCreaturePermanent(SubType.DWARF);
+        List<Permanent> dwarves = game.getBattlefield().getAllActivePermanents(dwarfFilter, controller.getId(), game);
+
+        FilterControlledPermanent equipmentFilter = new FilterControlledPermanent(SubType.EQUIPMENT);
+        List<Permanent> equipment = game.getBattlefield().getAllActivePermanents(equipmentFilter, controller.getId(), game);
+
+        if (!dwarves.isEmpty() && !equipment.isEmpty()) {
+            TargetPermanent target = new TargetPermanent(0, 1, dwarfFilter, true);
+            controller.choose(outcome, target, source.getId(), game);
+            Permanent dwarf = game.getPermanent(target.getFirstTarget());
+            if (dwarf != null) {
+                target = new TargetPermanent(0, Integer.MAX_VALUE, equipmentFilter, true);
+                controller.choose(outcome, target, source.getId(), game);
+                for (UUID targetId : target.getTargets()) {
+                    dwarf.addAttachment(targetId, source, game);
```

Comment:
It can attach multiple permanents. So the best way -- inform users about it. Example:
```
game.informPlayers(attachment.getLogName() + " was attached to " + sourcePermanent.getLogName());
```